persist/fs/retriever.go

```diff
@@ -313,6 +319,15 @@ func (r *blockRetriever) fetchBatch(
 
 		req.onRetrieved(seg)
 	}
+
+	err = seekerMgr.Return(shard, blockStart, seeker)
+	if err != nil {
+		r.logger.WithFields(
+			log.NewField("shard", shard),
+			log.NewField("blockStart", blockStart.Unix()),
+			log.NewField("err", err),
```

Comment:
nit: Might be worth `err.Error()` here on account of this coming out usually as a pointer value when printed instead of the actual err string.